pallets/xyk/src/lib.rs

```diff
@@ -503,53 +551,62 @@ impl<T: Trait> Module<T> {
 
         result
             .saturated_into::<u128>()
-            .saturated_into::<T::Balance>()
+            .saturated_into::<BalanceOf<T>>()
     }
 
     pub fn get_liquidity_asset(
-        first_asset_id: T::AssetId,
-        second_asset_id: T::AssetId,
-    ) -> T::AssetId {
-        if <LiquidityAssets<T>>::contains_key((first_asset_id, second_asset_id)) {
-            <LiquidityAssets<T>>::get((first_asset_id, second_asset_id))
-        } else {
-            <LiquidityAssets<T>>::get((second_asset_id, first_asset_id))
-        }
+        first_asset_id: CurrencyIdOf<T>,
```

Comment:
I do not think that we should use this approach, as get I think will return the default value if it doesn't find anything. So we definitely can't use that.